assets/app/scripts/services/applicationGenerator.js

```diff
@@ -231,8 +222,8 @@ angular.module("openshiftConsole")
           },
           source: {
             git: {
-              ref: sourceRef,
-              uri: sourceUrl
+              ref: input.buildConfig.gitRef || 'master',
```

Comment:
if they paste in something with a fragment, remove it and populate the ref field with it if empty?